modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/RegionClassLoader.java

```diff
@@ -339,19 +340,30 @@ public final URL findResource(final String name) {
   @Override
   public final Enumeration<URL> findResources(final String name) throws IOException {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. ([squid:S3776](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7561&issues=AWhqiKMiotIXnmsUkVkL&open=AWhqiKMiotIXnmsUkVkL)

[ISSUE_KEY]: AWhqiKMiotIXnmsUkVkL